diff --git a/lib/utils/searchUtils.js b/lib/utils/searchUtils.js
index 298fa256922abc626cd5a04a5cc2b686fb1867ca..910131ce57fbbc9ac7231ff9b0622618ce6f4023 100755
--- a/lib/utils/searchUtils.js
+++ b/lib/utils/searchUtils.js
@@ -37,7 +37,7 @@ export function useContextualSearchFilters() {
     }
     const tags = [
         DEFAULT_SEARCH_TAG,
-        ...Object.keys(allDocsData).map(getDocPluginTags),
+        ...(activePluginAndVersion ? [activePluginAndVersion.activePlugin.pluginId] : Object.keys(allDocsData)).map(getDocPluginTags),
     ];
     return {
         locale: i18n.currentLocale,
diff --git a/src/utils/searchUtils.ts b/src/utils/searchUtils.ts
index 082f0fb7085ca191a03360bff0bdb3758c046db7..9151b9691e7ab9954407b73bc8f51f0a2dc67c40 100755
--- a/src/utils/searchUtils.ts
+++ b/src/utils/searchUtils.ts
@@ -56,7 +56,7 @@ export function useContextualSearchFilters(): {locale: string; tags: string[]} {

   const tags = [
     DEFAULT_SEARCH_TAG,
-    ...Object.keys(allDocsData).map(getDocPluginTags),
+    ...(activePluginAndVersion ? [activePluginAndVersion.activePlugin.pluginId] : Object.keys(allDocsData)).map(getDocPluginTags),
   ];

   return {
